rename dev script, update contributor guide and readme (#9673)
sveltejs/svelte: CONTRIBUTING.md

### Installation

Ensure you have [pnpm](https://pnpm.io/installation) installed. After cloning the repository, run `pnpm install`.

### Developing

> To build the UMD version of `svelte/compiler` (this is only necessary for CommonJS consumers, or in-browser use), run `pnpm build` inside `packages/svelte`. To rebuild whenever source files change, run `pnpm dev`.

Reviewer: in-browser probably isn't exactly true since you can run ESM in the browser
```suggestion
To build the UMD version of `svelte/compiler` (this is only necessary for CommonJS consumers and the REPL), run `pnpm build` inside `packages/svelte`. To rebuild whenever source files change, run `pnpm dev`.
```

Reviewer 2: I think 'in-browser' is valid. I don't think there's a reasonable way currently to run an unbundled copy of the compiler in the browser.

> PREREQUISITE: Install chromium via playwright by running `pnpm playwright install chromium`

1. To run test, run `pnpm test`.
1. To run a particular test suite, use `pnpm test <suite-name>`, for example:

   ```bash
   pnpm test validator
   ```

> 1. To run an individual test _within_ a test suite, use `FILTER=<test-name> pnpm test <suite-name>`, for example:

Reviewer: Could we also document the vitest way of doing it? I find that easier to remember as it works on all my other projects as well (including SvelteKit)

   ```bash
   pnpm test validator -- -t a11y-alt-text
   ```

##### Updating `.expected` files

1. Tests suites like `snapshot` and `parser` assert that the generated output matches the existing snapshot.

> 1. To update these snapshots, run `UPDATE_SNAPSHOTS=true pnpm test`.

Reviewer: Can we just do `pnpm run test -u`? I just delegate to `vitest` that way in `enhanced-img`, but not sure if there's a special reason why that won't work here

Author: No, this is using a completely different mechanism. I have no idea how Vitest snapshots work. If it makes sense for us to use them then we can, though it'd be a different PR

Author: ok i just read the vitest snapshot docs and yeah, i don't think it's appropriate for our `snapshot` tests. it would make life harder
